ui/jstests/test_manage_taxonomies.jsx

```diff
@@ -77,7 +247,91 @@ define(['QUnit', 'jquery', 'setup_manage_taxonomies', 'reactaddons',
           done();
         });
       };
+      var vocabularyComponentRendered = React.addons.TestUtils.
+        renderIntoDocument(
+          <VocabularyComponent
+            vocabulary={vocabulary}
+            terms={vocabulary.terms}
+            reportError={reportError}
+            addTerm={addTerm}
+            repoSlug="repo"
+            ref={afterMount}
+          />
+        );
+      // check term title
+      var listOfLinks = React.addons.TestUtils.
+        scryRenderedDOMComponentsWithTag(
+          vocabularyComponentRendered,
+          'a'
+      );
+      assert.equal(listOfLinks.length, 2);
+      var linkHeader = listOfLinks[0].getDOMNode();
+      assert.equal(linkHeader.innerHTML, vocabulary.name);
 
+      //test items
+      var devItems = React.addons.TestUtils.
+        findRenderedDOMComponentWithClass(
+          vocabularyComponentRendered,
+          'panel-body'
+      );
+      var itemList = React.addons.TestUtils.
+        scryRenderedDOMComponentsWithTag(
+          devItems,
+          'li'
+      );
+      assert.equal(itemList.length, 2);
+      //test enter text in input text
+      var inputNode = React.addons.TestUtils.
+        findRenderedDOMComponentWithTag(
+          vocabularyComponentRendered,
+          'input'
+      );
+      React.addons.TestUtils.Simulate.change(
+        inputNode,
+        {target: {value: 'test12'}}
+      );
+      assert.equal(
```

Comment:
You will probably need to move this assertion inside `forceUpdate()` as described above